tests/core/resource/test_record.py

```diff
@@ -228,9 +228,69 @@ def test_patch_record_returns_updated_fields(self):
         self.assertEquals(self.stored['id'], self.result['id'])
         self.assertEquals(self.result['position'], 10)
 
+    def test_record_timestamp_is_not_updated_if_none_for_missing_field(self):
+        self.resource.request.json = {'data': {'polo': None}}
+        result = self.resource.patch()['data']
+        self.assertEquals(self.result['last_modified'],
+                          result['last_modified'])
+
+    def test_record_timestamp_is_not_updated_if_no_field_changed(self):
+        self.resource.request.json = {'data': {'position': 10}}
+        result = self.resource.patch()['data']
+        self.assertEquals(self.result['last_modified'],
+                          result['last_modified'])
+
+    def test_collection_timestamp_is_not_updated_if_no_field_changed(self):
+        self.resource.request.json = {'data': {'position': 10}}
+        self.resource.patch()
+        self.resource = self.resource_class(request=self.get_request(),
+                                            context=self.get_context())
+        self.resource.collection_get()['data']
+        last_modified = int(self.last_response.headers['ETag'][1:-1])
+        self.assertEquals(self.result['last_modified'], last_modified)
+
+    def test_timestamp_is_not_updated_if_no_change_after_preprocessed(self):
+        with mock.patch.object(self.resource, 'process_record') as mocked:
+            mocked.return_value = self.result
+            self.resource.request.json = {'data': {'position': 20}}
+            result = self.resource.patch()['data']
+            self.assertEquals(self.result['last_modified'],
+                              result['last_modified'])
+
+    def test_returns_changed_fields_among_provided_if_behaviour_is_diff(self):
+        self.resource.request.json = {'data': {'unread': True, 'position': 15}}
+        self.resource.request.headers['Response-Behavior'] = 'diff'
+        with mock.patch.object(self.resource.model, 'update_record',
+                               return_value={'unread': True, 'position': 0}):
+            result = self.resource.patch()['data']
+        self.assertDictEqual(result, {'position': 0})
+
+    def test_returns_changed_fields_if_behaviour_is_light(self):
+        self.resource.request.json = {'data': {'unread': True, 'position': 15}}
+        self.resource.request.headers['Response-Behavior'] = 'light'
+        with mock.patch.object(self.resource.model, 'update_record',
+                               return_value={'unread': True, 'position': 0}):
+            result = self.resource.patch()['data']
+        self.assertDictEqual(result, {'unread': True, 'position': 0})
```

Comment:
I just wrapped the Merge Patch tests in a new class, that moved up all the tests that were below. It's really weird on the diff.